ghost/core/core/frontend/helpers/get.js

```diff
@@ -131,6 +132,53 @@ function parseOptions(globals, data, options) {
  */
 async function makeAPICall(resource, controllerName, action, apiOptions) {
     const controller = api[controllerName];
+    let makeRequest = () => controller[action](apiOptions);
+
+    // Only do the optimisation on posts
+    if (resource === 'posts') {
+        try {
+            const parsedFilter = nqlLang.parse(apiOptions.filter);
+            // Support either `id:blah` or `id:blah+other:stuff`
+            if (parsedFilter.$and || parsedFilter.id) {
+                const queries = parsedFilter.$and || [parsedFilter.id];
+
+                for (const query of queries) {
+                    if ('id' in query) {
+                        if ('$ne' in query.id) {
+                            // This checks that there is only one occurence of a negative id filter
+                            // So we know it's safe to use the `replace` method
+                            if (apiOptions.filter.split('id:-').length === 2) {
+                                const idToFilter = query.id.$ne;
+
+                                // The default limit is 15, the addition order is to cast apiOptions.limit to a number
+                                const limit = apiOptions.limit ? 1 + apiOptions.limit : 16;
```

Comment:
Where does that default limit come from?

Reply:
That's a great question, I can't actually find where - but it is our default limit across our apis

Reply:
Yup - get a big plus one from me on that :D